boards/posix/native_posix/hw_models_top.c

```diff
@@ -134,13 +152,18 @@ void hwm_main_loop(void)
 		case IRQCNT:
 			hw_irq_ctrl_timer_triggered();
 			break;
+		case SDLEVENTTIMER:
+			break;
 		default:
 			/* LCOV_EXCL_START */
 			posix_print_error_and_exit(
 					"next_timer_index corrupted\n");
 			break;
 			/* LCOV_EXCL_STOP */
 		}
+#ifdef CONFIG_HAS_SDL
+		sdl_handle_synchronious_events();
```

Comment:
Please do not set code to execute in each delta cycle. You do not need to and it would bog down everything else.

Reply:
I could move it into the case, but my idea here was that you would like to act as fast as possible on the asynchronous events and as such check it for every expired timer.